mqtt/mqtt-broker/src/broker.rs

```diff
@@ -701,6 +700,7 @@ where
 
     fn open_session(&mut self, auth_id: AuthId, connreq: ConnReq) -> Result<OpenSession, Error> {
         let client_id = connreq.client_id().clone();
+        let client_info = ClientInfo::new(connreq.peer_addr(), auth_id.clone());
```

Comment:
lets move this line closer to usage?

Reply:
We use client_info within the match block multiple times. We could recreate it a few times if that's better, but I prefer like this